addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -7978,6 +7978,15 @@ registry.SnippetSave = SnippetOptionWidget.extend({
                                     const defaultSnippetName = _.str.sprintf(_t("Custom %s"), this.data.snippetName);
                                     const targetCopyEl = this.$target[0].cloneNode(true);
                                     targetCopyEl.classList.add('s_custom_snippet');
+                                    // when cloning the snippets which has o_snippet_invisible, o_snippet_mobile_invisible or
+                                    // o_snippet_desktop_invisible class will be hidden because of d-none class added on it,
+                                    // so we needs to remove `d-none` explicity in such case from the target.
+                                    const isTargetHidden = !(!this.$target.hasClass('o_snippet_invisible')
+                                        && !this.$target.hasClass('o_snippet_mobile_invisible')
+                                        && !this.$target.hasClass('o_snippet_desktop_invisible'));
```

Comment:
Don't use jQuery methods. this.$target[0] represents the HTMLElement you want to test, and don't forget to use double quotes for strings in JS. So here a suggestion:

```suggestion
                                    const isTargetHidden = [
                                        "o_snippet_invisible",
                                        "o_snippet_mobile_invisible",
                                        "o_snippet_desktop_invisible"
                                    ].some(className => this.$target[0].classList.contains(className));
```

Reply:
Hello @SergeBayet,
Thank you for bringing this to our attention. We have made the necessary changes—could you please review them?
Thanks